File: Lava/hello.hs
module Main where

import Lava
import Control.Monad (liftM)

-- Simple Hello World circuit in Lava
helloWorld :: Signal Bool -> Signal Bool
helloWorld input = output
  where
    output = input

-- Test circuit that outputs "Hello, World!" pattern
main :: IO ()
main = do
    putStrLn "Hello, World! (Lava HDL)"
    
    -- Simple circuit simulation
    let testInput = high
    let result = helloWorld testInput
    
    -- In a real Lava program, this would generate hardware
    putStrLn "Lava circuit generated: Hello, World!"

File: MyHDL/hello.py
from myhdl import *

@block
def hello_world():
    """A simple Hello World module in MyHDL"""
    
    @always_comb
    def logic():
        print("Hello, World! (MyHDL)")
    
    return logic

# Test the module
if __name__ == "__main__":
    hello_inst = hello_world()
    hello_inst.run_sim()
    print("Hello, World! - MyHDL simulation completed")